### PR TITLE
Funcy list computation

### PR DESCRIPTION
**Pull request by Gab-km**

### CODE REVIEW

File: Funcy.Test/ApplicativeLawsCheck.fs [hunk 1]
```diff
@@ -164,3 +164,80 @@ module ApplicativeLawsCheck =
...
+
+        let ``Identity in Nil<T>`` = Prop.forAll(Arb.int)(fun i ->
+            let v = FuncyList<int>.Nil()
+            // pure id <*> v = v
+            v.Apply(pureFList funcId) = (v :> FuncyList<int>)
+        )
+
+        let ``Composition in FuncyList<T> 1`` = Prop.forAll(Arb.systemFunc(CoArbitrary.int, Arb.int), Arb.systemFunc(CoArbitrary.int, Arb.int), Arb.int)(fun f g i ->
+            let u = FuncyList.Cons(f, FuncyList.Nil())
+            let v = FuncyList.Construct([|g|])
```
Comment (cocoatomo) [inline Funcy.Test/ApplicativeLawsCheck.fs]: Are there some intention to use two constructors: "FuncyList.Cons(f, FuncyList.Nil())" and "FuncyList.Construct([|g|])"?

Comment (Gab-km) [inline Funcy.Test/ApplicativeLawsCheck.fs]: These are to include diversity of types.
Cons returns Cons type value, Nil returns Nil type value, and Construct returns FuncyList type value.
I want to check the Functor laws with various types. :smile:

Comment (cocoatomo) [inline Funcy.Test/ApplicativeLawsCheck.fs]: :+1:

**Comment by cocoatomo** [comment]
LGTM!

**Comment by Gab-km** [comment]
Thanks!

**Pull request merged**